app-frontend/src/app/components/pages/project/layer/aoi/index.js

```diff
@@ -1,11 +1,256 @@
+        this.projectService
+            .updateProjectLayer(updatedLayer)
+            .then(newLayer => {
+                this.layer = newLayer;
+                this.layerGeom = this.layer.geometry;
+                if (!isSaveShape) {
+                    this.$state.go('project.layer');
+                } else {
+                    this.openShapeModal(this.layer);
+                }
+            })
+            .catch(err => {
+                this.$log.error(err);
+                this.$window.alert('There is an error updating this layer\'s AOI.'
+                    + ' Please try again later');
+            });
+    }
+
+    formatDateDisplay(date) {
+        return date.length ? this.moment.utc(date).format('LLL') + ' (UTC)' : 'MM/DD/YYYY';
+    }
+
+    openShapeModal(layer) {
+        const modal = this.modalService.open({
+            component: 'rfEnterTokenModal',
+            resolve: {
+                title: () => 'Enter a name for the vector data',
+                token: () => `${layer.name} AOI `
+                    + `- ${this.formatDateDisplay(layer.createdAt)}`
+            }
+        });
+
+        modal.result.then((name) => {
+            this.shapesService
+                .createShape(this.generateShapeCreate(name, layer.geometry))
+                .then(() => {
+                    this.$state.go('project.layer');
+                })
+                .catch(err => {
+                    this.$window.alert('There is an error adding this layer\'s AOI as vector data.'
+                        + ' Please try again later');
+                    this.$log.error(err);
+                });
+        }).catch(() => {});
+    }
+
+    generateShapeCreate(name, geometry) {
+        return {
+            type: 'FeatureCollection',
+            features: [
+                {
+                    type: 'Feature',
+                    properties: { name },
+                    geometry,
+                    id: this.uuid4.generate()
+                }
+            ]
+        };
+    }
+
+    onCancelDrawAoi() {
+        this.$state.go('project.layer');
+    }
+
+    onChangeFilterList(filterId) {
+        this.fetchPage(1, filterId);
+        this.displayAoiSource = this.filterList.find(fl => fl.id === filterId).name;
+    }
+
+    createItemInfo(fl) {
+        if (this.dataType === 'AOI') {
+            return {
+                id: fl.id,
+                name: fl.name,
+                subtext: '',
+                date: fl.createdAt,
+                colorGroupHex: fl.colorGroupHex,
+                geometry: fl.geometry
+            };
+        }
+
+        if (this.dataType === 'VECTOR') {
+            return {
+                id: fl.id,
+                name: fl.properties.name,
+                subtext: '',
+                date: fl.properties.createdAt,
+                colorGroupHex: defaultColor,
+                geometry: fl.geometry
+            };
+        }
+
+        return 0;
```

Comment:
@Lknechtli  This is for excluding the unsupported data type that don't match `this.filterList`, so that those ones won't list on the sidebar for selection since `_.compact()` is being applied. I could log an error here but not an alert.

Reply:
hm, I think I'd advocate for just filtering them out instead of using falsy values in that case, like 
```javascript
this.itemList = results.filter(
	(item) => ['aoi', 'vector'].includes(item.dataType.toLowerCase())
).map(this.createItemInfo.bind(this));
```